src/main/java/Product.java

```diff
@@ -0,0 +1,16 @@
+public class Product extends Record{
+
+    private String category;
+
+    public Product (String name, float price, String date, String productType){
+        super( name, price, date);
+        this.category = productType;
+    }
+    public float getPrice(){
+        return this.price;
+    }
+
+    public String getCategory() {
+        return category;
+    }
+}
```

Comment:
Please add in getter and setter for all the classes and all the attributes! thank you! They will be needed.